lib/WWW/YouTube/Download.pm

```diff
@@ -442,6 +442,216 @@ sub user_id {
+	$ref->{contents}->{twoColumnBrowseResultsRenderer}->{tabs}->[0]->{tabRenderer}->{content}->{sectionListRenderer}->{contents}->[0]->{itemSectionRenderer} &&
+	$ref->{contents}->{twoColumnBrowseResultsRenderer}->{tabs}->[0]->{tabRenderer}->{content}->{sectionListRenderer}->{contents}->[0]->{itemSectionRenderer}->{contents} && ref($ref->{contents}->{twoColumnBrowseResultsRenderer}->{tabs}->[0]->{tabRenderer}->{content}->{sectionListRenderer}->{contents}->[0]->{itemSectionRenderer}->{contents}) eq 'ARRAY' &&
+	$ref->{contents}->{twoColumnBrowseResultsRenderer}->{tabs}->[0]->{tabRenderer}->{content}->{sectionListRenderer}->{contents}->[0]->{itemSectionRenderer}->{contents}->[0] &&
+	$ref->{contents}->{twoColumnBrowseResultsRenderer}->{tabs}->[0]->{tabRenderer}->{content}->{sectionListRenderer}->{contents}->[0]->{itemSectionRenderer}->{contents}->[0]->{playlistVideoListRenderer} &&
+	$ref->{contents}->{twoColumnBrowseResultsRenderer}->{tabs}->[0]->{tabRenderer}->{content}->{sectionListRenderer}->{contents}->[0]->{itemSectionRenderer}->{contents}->[0]->{playlistVideoListRenderer}->{contents} && ref($ref->{contents}->{twoColumnBrowseResultsRenderer}->{tabs}->[0]->{tabRenderer}->{content}->{sectionListRenderer}->{contents}->[0]->{itemSectionRenderer}->{contents}->[0]->{playlistVideoListRenderer}->{contents}) eq 'ARRAY'
+	;
+
+	my $playlistVideoListRenderer = $ref->{contents}->{twoColumnBrowseResultsRenderer}->{tabs}->[0]->{tabRenderer}->{content}->{sectionListRenderer}->{contents}->[0]->{itemSectionRenderer}->{contents}->[0]->{playlistVideoListRenderer};
+
+	## weed out videos
+	my @videos;
+	my $cnt = 0;
+	for my $renderer (@{ $playlistVideoListRenderer->{contents} }){
+		$cnt++;
+		my $video = $renderer->{playlistVideoRenderer};
+
+		push(@videos, {
+			id	=> $video->{videoId},
+			runtime => $video->{lengthSeconds},
+			title	=> ($video->{title}->{simpleText}||'undef'), # encoded in Perl internal
+		});
+
+		if($self->{verbose}){
+			print " $cnt id:". $video->{videoId};
+			print " runtime:". ($video->{lengthSeconds}||'undef');
+			print " title:". Encode::encode_utf8($video->{title}->{simpleText}||'undef');
+			print "\n";
+		}
+	}
+
+	## next page
+	my $nextUrl;
+	if(	$playlistVideoListRenderer->{continuations} && ref($playlistVideoListRenderer->{continuations}) eq 'ARRAY' &&
+		$playlistVideoListRenderer->{continuations}->[0] &&
+		$playlistVideoListRenderer->{continuations}->[0]->{nextContinuationData}
+	){
+		my $next = $playlistVideoListRenderer->{continuations}->[0]->{nextContinuationData};
+		my $continuation = $next->{continuation};
+		my $ctoken = $continuation;
+		my $itct = $next->{clickTrackingParams};
+
+		if($self->{verbose}){
+			print " next page continuation data:\n cont:". $continuation ."\n";
+			print " itct:". $itct ."\n";
+			print "\n";
+		}
+
+		# $nextUrl = 'https://www.youtube.com/browse_ajax?action_continuation=1&continuation='. $continuation;
```

Comment:
Do we need the commented code?

Reply:
Agreed. The updted version has this comment removed.